gobblin-core/src/main/java/gobblin/source/extractor/partition/Partitioner.java

```diff
@@ -33,26 +39,49 @@
 import gobblin.source.extractor.watermark.WatermarkPredicate;
 import gobblin.source.extractor.watermark.WatermarkType;
 
-
 /**
  * An implementation of default partitioner for all types of sources
  */
 public class Partitioner {
   private static final String WATERMARKTIMEFORMAT = "yyyyMMddHHmmss";
   private static final Logger LOG = LoggerFactory.getLogger(Partitioner.class);
+  public static final String HAS_USER_SPECIFIED_HIGH_WATERMARK = "has.user.specified.high.watermark";
```

Comment:
Add a namespace and use camel case, so something like "partitioner.hasUserSpecifiedHighWatermark".